surrealdb.ts

```diff
@@ -0,0 +1,67 @@
+// surrealDB adapter for telegraf session
+
+import * as Surreal from "surrealdb.js";
+import { SessionStore } from "./types";
+import {defaults} from "./defaults";
+
+
+type TSurrealRecord = {
+	id: string;
+	value: any;
+}
+interface NewClientOpts {
+	url?: string;
+	auth?: Surreal.RootAuth;
+	namespace?: string;
+	database?: string;
+	onInitError?: (err: unknown) => void;
+}
+
+interface ExistingClientOpts {
+	client: Surreal.default;
+}
+
+export function SurrealDB<Session>(opts: NewClientOpts): SessionStore<Session>;
+export function SurrealDB<Session>(opts: ExistingClientOpts): SessionStore<Session>;
+export function SurrealDB<Session>(opts: NewClientOpts | ExistingClientOpts): SessionStore<Session> {
+	interface SessionDoc {
+		key: string;
+		session: Session;
+	}
+
+	let client: Surreal.default;
+
+	if ("client" in opts) client = opts.client;
+	else {
+		client = new Surreal.default(`${opts.url}/rpc`);
+		client.signin(opts.auth!)
+			.then(async () => {
+				await client.use(opts.namespace!, opts.database!);
+			})
+			.catch(opts.onInitError!);
+	}
+
+	// Small changes associated with the naming rules:
+	// - replacement "-" on "_" in the name of the base,
+	// - replacement ":" on "_" in the identifier.
+	const surrealDefaultTable = defaults.table.replace(/-/g, '_');
+	const surrealId = (key: string): string => `${surrealDefaultTable}:${key.replace(/:/g, '_')}`;
+
+	return {
+		async get(key) {
+			await client.wait();
```

Comment:
Move `client.wait()` before return and assign it to a variable. Then await that variable here. Also errors should be caught and passed to opts.onInitError:

```TS
const conn = client.wait();
conn.catch(opts.onInitError);
```

I understand the Redis adapter is missing this as well; I'll fix it. Meanwhile, refer to the MongoDB adapter.